src/components/shared/inputs/input-shell.jsx

```diff
@@ -0,0 +1,49 @@
+import clsx from 'clsx';
+import styles from './shared-inputs.module.scss';
+
+export default function InputShell({
+  type = '',
+  name,
+  label,
+  errors = '',
+  register,
+  classNames = '',
+  required = false,
+}) {
+  const inputType = {
+    input: (
+      <input
+        className={styles.input}
+        placeholder='Type text'
+        type='text'
+        {...register(name, { required: required })}
+      />
+    ),
+    textarea: (
+      <textarea
+        className={styles.textarea}
+        placeholder='Type a message'
+        type='text'
+        {...register(name, { required: required })}
+      />
+    ),
+  };
+
+  return (
+    <div
+      className={clsx(
+        styles.formGroup,
+        classNames.gridArea,
+        errors.formError ? styles.activeError : ''
+      )}
+    >
+      <label className={styles.label}>{label}</label>
+
+      {inputType[type]}
```

Comment:
This inputType object returns a React component, and you always want to use React syntax when rendering JSX. 

```
const InputComponent = inputType[type]
```
And then in your return:
```
<InputComponent />
```